tests/recorder.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+/*
+    Usage:
+
+        node recorder --test={#} --description="This test describes {xyz} conditions"
+
+    Requirements:
+
+        -  The `test{#}` directory must exist within the `fixtures` directory
+        -  The `test{#}` directory must contain `payload.json` and `headers.json` files, representing info POSTed form GitHub
+
+*/
+
+var nock = require('nock');
+var fs = require('fs');
+var request = require('request');
+var yargs = require('yargs').argv;
+
+var testUtils = require('./test-utils');
+var logger = require('../logger');
+
+// Run the server
+require('../bin/www');
+// Process Redis tasks
+require('../worker');
+
+var timeout = 10000;  // 10 seconds
+var testNumber;
+
+// Require that a user provide the test number (which represents the fixtures/{testno})
+// via the command line.  If none provided, bail.
+if(yargs.test && !isNaN(yargs.test)) {
+    testNumber = yargs.test;
+}
+else {
+    logAndQuit('No test provided, exiting.', process.argv);
+}
+
+if(!yargs.description) {
+    logAndQuit('No description provided, exiting.', process.argv);
+}
+
+// Start recording the requests sent to GitHub as well as the responses sent back
+nock.recorder.rec({ output_objects: true });
+
+// Kick off!  Simulates us receiving a POST from GitHub!
+request.post({
+    url: testUtils.appHost + '/hook',
+    headers: testUtils.getFileContents(testNumber, 'headers'),
+    form: testUtils.getFileContents(testNumber, 'payload')
+}, function(error) {
+    if(error) logAndQuit('Post for kicking off the recording could not be completed: ', error);
```

Comment:
Ah, `logAndQuit` uses `process.exit`. So we don't need to worry about returning here.